filters/include/pcl/filters/impl/voxel_grid_covariance.hpp

```diff
@@ -352,6 +352,7 @@ pcl::VoxelGridCovariance<PointT>::applyFilter (PointCloud &output)
         leaf.cov_ = leaf.evecs_ * eigen_val * leaf.evecs_.inverse ();
       }
       leaf.evals_ = eigen_val.diagonal ();
+      leaf.cov_det_ = leaf.cov_.determinant ();
 
       leaf.icov_ = leaf.cov_.inverse ();
       if (leaf.icov_.maxCoeff () == std::numeric_limits<float>::infinity ( )
```

Comment:
To me it is not clear why this condition is necessary at all. Since the original covariance matrix is real and symmetric, `evecs_` is orthogonal. The coefficients of the original covariance matrix are limited due to limited voxel size. Lines 333-352 ensure that the covariance matrix is non-singular and not even close to singularity. What other reason could there be for a coefficient of the inverse to be inf or -inf?